distributed/tests/test_profile.py

```diff
@@ -211,27 +211,29 @@ def stop():
 def test_watch_requires_lock_to_run():
     start = time()
 
-    def stop_lock():
-        return time() > start + 0.600
+    release_lock = False
 
-    def stop_profile():
+    def stop_blocking():
+        return release_lock
```

Comment:
Changing this from a timing-based condition to an explicitly set one _after_ the test criterion is evaluated is the main change, the rest is cosmetic.